source/extensions/tracers/datadog/datadog_tracer_impl.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <datadog/opentracing.h>
+
+#include "envoy/local_info/local_info.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/thread_local/thread_local.h"
+#include "envoy/tracing/http_tracer.h"
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/http/header_map_impl.h"
+#include "common/json/json_loader.h"
+
+#include "extensions/tracers/common/ot/opentracing_driver_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+#define DATADOG_TRACER_STATS(COUNTER)                                                              \
+  COUNTER(traces_sent)                                                                             \
+  COUNTER(timer_flushed)                                                                           \
+  COUNTER(reports_sent)                                                                            \
+  COUNTER(reports_dropped)                                                                         \
+  COUNTER(reports_failed)
+
+struct DatadogTracerStats {
+  DATADOG_TRACER_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+class TraceReporter;
+typedef std::unique_ptr<TraceReporter> TraceReporterPtr;
+typedef std::shared_ptr<datadog::opentracing::TraceEncoder> TraceEncoderPtr;
+
+/**
+ * Class for a Datadog-specific Driver.
+ */
+class Driver : public Common::Ot::OpenTracingDriver {
+public:
+  /**
+   * Constructor. It adds itself and a newly-created Datadog::Tracer object to a thread-local store.
+   */
+  Driver(const envoy::config::trace::v2::DatadogConfig& datadog_config,
+         Upstream::ClusterManager& cluster_manager, Stats::Store& stats,
+         ThreadLocal::SlotAllocator& tls, Runtime::Loader& runtime);
+
+  // Tracer::OpenTracingDriver
+  opentracing::Tracer& tracer() override;
+  PropagationMode propagationMode() const override {
+    return Common::Ot::OpenTracingDriver::PropagationMode::TracerNative;
+  }
+
+  // Getters to return the DatadogDriver's key members.
+  Upstream::ClusterManager& clusterManager() { return cm_; }
+  Upstream::ClusterInfoConstSharedPtr cluster() { return cluster_; }
+  Runtime::Loader& runtime() { return runtime_; }
+  DatadogTracerStats& tracerStats() { return tracer_stats_; }
+  const datadog::opentracing::TracerOptions& tracerOptions() { return tracer_options_; }
```

Comment:
I think there's an Envoy convention for putting these non-overridden methods before the overridden ones.